### DISCUSSION TITLE
Subscribed react components break after calling `set` after awaiting within an async write atom.

### DISCUSSION THREAD

**Discussion by mwildehahn** [pmndrs/jotai#2454] [Bug report] [answered]
I have an atom that looks something like:

```
export const sendMessageAtom = atom(
  null,
  async (get, set, message: Pick<OptimisticMessage, "content">) => {
    set(optimisticMessagesByIdAtom, (optimisticMessagesById) => ({
      ...optimisticMessagesById,
      [optimisticMessage.id]: optimisticMessage,
    }));

    const { message: sentMessage } = await fetchApi<{
      ok: boolean;
      message?: Message;
    }>("/messages, {
      method: "POST",
      body: JSON.stringify({
        text: message.content.text,
        client_id: optimisticMessage.id,
      }),
    });

    if (sentMessage) {
      // once this gets invoked, the components no longer update?
      set(receiveMessageAtom, sentMessage);
    }
  }
);
```

Calling `set(receiveMessageAtom, sentMessage);` after awaiting the promise causes the components to no longer update when values are changed. I tested with some dummy values and invoking it before awaiting the promise works fine. If I comment out setting the value after the promise, the components stay updated. As soon as that is invoked though, the jotai state will get updated but the UI components will no longer re-render.

Any ideas? I'm super confused.

**Comment by mwildehahn** [comment]
I tried re-creating this issue in codesandbox and couldn't reproduce. Is there a way to debug what components are subscribed?

**Comment by dai-shi** [comment]
Does it work with v2.6.5?

**Comment by mwildehahn** [comment]
i upgraded to `2.7.0` as a part of debugging this and it isn't working, let me try 2.6.5

**Comment by mwildehahn** [comment]
ah man. it does work with 2.6.5 😭 .

**Reply by dai-shi** [reply, marked as answer]
It must be a regression with #2396.

Can you try #2451?
https://ci.codesandbox.io/status/pmndrs/jotai/pr/2451
☝️ Find "Local Install Instructions"

**Reply by mwildehahn** [reply]
Yes this works! Thank you for the prompt reply. I was refactoring this code path today and have spent like 6 hours debugging because I thought my refactor somehow caused this 😅 . I was going to quit programming if I couldn't figure out what was wrong.

Thanks for the lib and everything you do ❤️

**Reply by dai-shi** [reply]
Thanks for confirming.
